encrypted-pii/backend/src/main/java/org/phenotips/data/encrypted/internal/PIISearchStubSync.java

```diff
@@ -47,6 +48,39 @@ public PIISearchStubSync()
         super("patient-birthyear-updater", new PatientChangingEvent());
     }
 
+    private void readDates(BaseObject obj, String fromDateProperty, String toYearProperty) {
+        String enteredValue = obj.getStringValue(fromDateProperty);
+        if (enteredValue != null && !enteredValue.isEmpty()) {
+            JSONObject enteredJSON = new JSONObject(enteredValue);
+
+            String year = enteredJSON.optString("year");
+            JSONObject rangeJSON = enteredJSON.optJSONObject("range");
+
+            if (rangeJSON != null && year != null && !year.isEmpty()) {
```

Comment:
Use `StringUtils.isNotEmpty(year)`. Same below for `yearRange` and `year` further down.